docs/BUILD.md

```diff
@@ -0,0 +1,263 @@
+# Building s2n-tls
+
+Obtain s2n-tls by cloning the Github repo locally:
+```bash
+git clone https://github.com/aws/s2n-tls.git
+cd s2n-tls
+```
+
+## Building
+
+s2n-tls can be built as follows:
+
+<details open>
+<summary>Ubuntu</summary>
+
+```bash
+# install build dependencies
+sudo apt update
+sudo apt install cmake ninja-build
+
+# install a libcrypto
+sudo apt install libssl-dev
+
+# build s2n-tls
+cmake . -Bbuild -GNinja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=./s2n-tls-install
+ninja -C build -j $(nproc)
+CTEST_PARALLEL_LEVEL=$(nproc) ninja -C build test
+ninja -C build install
+```
+</details>
+
+<details>
+<summary>MacOS</summary>
+
+```bash
+# install build dependencies
+brew install cmake ninja
+
+# install a libcrypto
+brew install openssl@3
+
+# build s2n-tls
+cmake . -Bbuild -GNinja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_PREFIX_PATH=$(dirname $(dirname $(brew list openssl@3|grep libcrypto.dylib))) \
+    -DCMAKE_INSTALL_PREFIX=./s2n-tls-install
+ninja -C build -j $(sysctl -n hw.ncpu)
+CTEST_PARALLEL_LEVEL=$(sysctl -n hw.ncpu) ninja -C build test
+ninja -C build install
```

Comment:
Have you tested all these builds?

I'm also not sure ninja should be our default build suggestion. Don't we primarily use cmake directly?

Reply:
I agree, does anyone on our team actually use ninja? I don't.

Reply:
I switched the normal examples to pure CMake and I made the non-ninja section a ninja section.